provider/clusterrosaclassic/cluster_rosa_classic_state.go

```diff
@@ -99,3 +107,48 @@ type PrivateHostedZone struct {
 	ID      types.String `tfsdk:"id"`
 	RoleARN types.String `tfsdk:"role_arn"`
 }
+
+func flattenAdminCredentials(username string, password string) types.Object {
+	attributeTypes := map[string]attr.Type{
+		"username": types.StringType,
+		"password": types.StringType,
+	}
+	if username == "" && password == "" {
+		return types.ObjectNull(attributeTypes)
+	}
+
+	attrs := map[string]attr.Value{
+		"username": types.StringValue(username),
+		"password": types.StringValue(password),
+	}
+
+	return types.ObjectValueMust(attributeTypes, attrs)
+}
+
+func expandAdminCredentials(ctx context.Context, object types.Object, diags diag.Diagnostics) (username string, password string) {
+	if object.IsNull() {
+		return "", ""
+	}
+
+	var conf AdminCredentials
+	diags.Append(object.As(ctx, &conf, basetypes.ObjectAsOptions{})...)
+	if diags.HasError() {
+		return "", ""
+	}
+
+	return conf.Username.ValueString(), conf.Password.ValueString()
+}
+
+func adminCredentialsNull() types.Object {
+	return flattenAdminCredentials("", "")
+}
+
+func adminCredentialsEqual(state, plan types.Object) bool {
```

Comment:
The same comment here - 
I think you can use the ValidateStateAndPlanEquals for that since it verified `!stateAttr.Equal(planAttr) ` please check it and remove it if it already covered....

Reply:
@bardielle thanks for the comments
per my understanding, `ValidateStateAndPlanEquals` is not useful for the complex object so it's not used in existing code, but use reflect.DeepEqual() like https://github.com/terraform-redhat/terraform-provider-rhcs/blob/8b3e60f7fee4608e925b3c87a4c631d92d6461dd/provider/clusterrosaclassic/cluster_rosa_classic_resource.go#L1024-L1029

Also, when comparing AdminCredential, we need to handle the case that state is nil&plan is unknown, which should not be recognized as invalid. That's why I add the `adminCredentialsEqual` func
let me know if you have any concern on this. Thanks

Reply:
> Also, when comparing AdminCredential, we need to handle the case that state is nil&plan is unknown

This is not true any more, as you use the UseStateIfUnknown plan modifier, no?
So if the state is null and the plan is unknown, this function should already be called with plan=null

Reply:
Did you test the `ValidateStateAndPlanEquals` function with adminCredentials? I think it's worth to check.
This function has been called from the update function so it means the attribute will have a value

Reply:
@bardielle @nirarg 
Thanks for the comments. I tried to use `ValidateStateAndPlanEquals` to compare admincredentials in state&plan, it works, I updated my commit, please help to re-review.

Reply:
@bardielle @nirarg 
I tried `ValidateStateAndPlanEquals`, and it could not work.
my test case:
created a cluster w/o cluster admin user created, then changed the main.tf with updated rhcs_cluster_rosa_classic.properties, then I will run into the error when issue `terraform apply`
```
rhcs_cluster_rosa_classic.rosa_sts_cluster: Modifying... [id=29c3pc9j7258jj5i6roc0atbol1i7qmj]
╷
│ Error: Attribute value cannot be changed
│ 
│   with rhcs_cluster_rosa_classic.rosa_sts_cluster,
│   on main.tf line 50, in resource "rhcs_cluster_rosa_classic" "rosa_sts_cluster":
│   50: resource "rhcs_cluster_rosa_classic" "rosa_sts_cluster" {
│ 
│ Attribute admin_credentials, cannot be changed from <null> to <unknown>
```

Also, all test cases in subsystem test related with update flow failed.

Reply:
Thank you for validating it. I'm approving the PR